src/core/datatypes/trees/Clade.cpp

```diff
@@ -728,3 +731,117 @@ void Clade::setAges(const std::vector<Taxon>& taxa)
         }
     }
 }
+
+// These should provide a "strict weak ordering" that returns
+// true only if c1 is ordered with respect to c2, and c1 < c2.
+// We do not treat conflicting constraints as unordered.
+//
+// Its not really clear how to handle negative constraints or optional constraints.
+// We really should separate the idea of constraints (CladeConstraint,
+// NotCladeConstraint, OptionalCladeConstraint) from the idea of clades.
+
+bool clade_before(const Clade& c1, const Clade& c2)
```

Comment:
Usually we use camel-case for our methods and class names. Maybe change this?

Reply:
Fixed.